docs/jargon-mesh.md

```diff
@@ -0,0 +1,658 @@
+---
+id: terminology
+title: Temporal's core terminology
+sidebar_label: Terminology
+---
+
+import Mermaid from '@theme/Mermaid';
+
+## Workflow
+
+A fault-oblivious stateful function that orchestrates Activities.
+
+- A Workflow has full control over which [**Activities**](#activity) are executed, and in which order.
+- A Workflow must not affect the external world directly, only through [**Activities**](#activity).
```

Comment:
IMO only, this is a pretty hard requirement as
we do have sections of SideEffects
I would lower this to a "recommendation" honestly